Added EDID tools for dumping and flashing screen data

radujipa opened the pull request:
Two tools have been added here: rpi-dump-edid is a dev & user facing
script which dumps the screen EDID to the /boot/ partition for testing
data and debugging. generate-rpi-edid-dumps is a tool which uses raw
EDIDs to generate test data for kano-settings display functions.
These will be the tools to build on the res/edid dataset.

NOTE: Might be worth reviewing the `generate-rpi-edid-dumps` tool without the diff and as a whole file.

radujipa commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Generate inferred EDID data from raw EDID dump for screen tests](https://trello.com/c/c6ss4uIO/1344-generate-inferred-edid-data-from-raw-edid-dump-for-screen-tests)

skarbat commented on bin/rpi-dump-edid:
```diff
@@ -0,0 +1,82 @@
+import os
+import sys
+
+from kano.utils.shell import run_cmd
```
Just a question. What should be the general use here for `run_cmd`? `kano.utils.shell` or `kano.utils` ?

radujipa replied:
Looking back at it, the only reason why we have so many `from kano.utils import` statements is because we opted to write a nasty import there for backwards compatibility reasons. 

I think providing the full import path is clearer since you know at a glance where you need to look if you want to see the implementation of that function.

skarbat replied:
Nice, yeah I agree. 👍

skarbat commented on bin/rpi-dump-edid:
```diff
@@ -0,0 +1,82 @@
+
+    md5sum = out.split(' ', 1)[0]
+    model_md5 = '{}-{}'.format(model, md5sum)
+
```
Should this bit be wrapped in a `try-except` just in case?

radujipa replied:
Good idea! Fixed

skarbat commented on tools/edid/generate-rpi-edid-dumps:
```diff
@@ -5,51 +5,82 @@
+#
+# WARNING: This tool can destroy the machine and display!
+#          See edid-rw README for the dangers of using it.
+#
```
Would it be better to move these important considerations into the docopt help section?

radujipa replied:
Good point! Done

skarbat commented on tools/edid/generate-rpi-edid-dumps:
```diff
@@ -70,107 +101,363 @@ def run_cmd(cmd):
+    screen_models_with_md5 = [
+        raw_edid[:-len(os.path.basename(raw_edid_path_pattern.format(filename='')))]
+        for raw_edid in os.listdir(raw_edid_dir)
+    ]
```
nice

radujipa replied:
An obscure one, I know... but reduces code maintenance

skarbat commented on tools/edid/generate-rpi-edid-dumps:
```diff
@@ -70,107 +101,363 @@ def run_cmd(cmd):
+
+def check_i2c_bus(bus_number):
+    """ Check that the i2c device at address 3f805000.i2c is available. """
+
```
This address is provided by `i2c_bcm2708` kernel module right?
Can we mention it here? Does it need a check using a `lsmod` or similar?

radujipa replied:
This one is required for the `edid-rw` tool to work properly on RPi. The i2c address to write to the HDMI port does not change. 

There are comments and a hint to use `i2cdetect -l` to check that it's available. What other checks would be necessary? (not sure I understand your questions)

skarbat replied:
Oh I understand now. I wasn't sure about the purpose of the address that's all.

skarbat commented on tools/edid/generate-rpi-edid-dumps:
```diff
@@ -70,107 +101,363 @@ def run_cmd(cmd):
-    if os.geteuid() == 0:
-        print 'Program must NOT be run as root.'
-        sys.exit(RC_NOT_USER)
+    if os.geteuid() != 0:
```
I am not sure that we need to `set-uid` this tool.

radujipa replied:
Yes, did that one again... Good spot! Fixed

skarbat commented:
Great work, Radu.

radujipa commented:
Thanks for the review, @skarbat 👍

tombettany commented on bin/rpi-dump-edid:
```diff
@@ -0,0 +1,86 @@
+
+    # If the EDID is corrupt, the proc might return '[E] No device present'.
+    if 'device_name' in out:
+        model = out.strip('device_name=').strip()
```
Given that we almost certainly have a kano-settings function to do read the name, do we want to use that instead?

radujipa replied:
Great point. That script started as a standalone tool, but forgot to integrate it with the system properly after I moved it to `bin/`. I'll look to move most of that functionality in kano-settings instead. Thanks for pointing that out 👍

radujipa commented:
@tombettany @skarbat 

As per Tom's suggestion, I removed the `rpi-dump-edid` and moved it into `kano-settings-cli` and the functionality in `kano_settings.system.display` instead.
